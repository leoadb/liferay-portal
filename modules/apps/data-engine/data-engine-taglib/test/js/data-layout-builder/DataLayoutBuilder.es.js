/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import DataLayoutBuilder from '../../../src/main/resources/META-INF/resources/data_layout_builder/js/data-layout-builder/DataLayoutBuilder.es';

let component;
const spritemap = 'icons.svg';

describe('DataLayoutBuilder', () => {
	beforeEach(() => {
		jest.useFakeTimers();
	});

	afterEach(() => {
		if (component && component.dispose) {
			component.dispose();
		}
	});

	it('is rendering', () => {
		component = new DataLayoutBuilder({
			readOnly: false,
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});
});
