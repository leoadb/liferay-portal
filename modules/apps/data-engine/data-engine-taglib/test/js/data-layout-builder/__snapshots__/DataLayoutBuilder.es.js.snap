// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataLayoutBuilder is rendering 1`] = `
DataLayoutBuilder {
  "containerRef": Object {
    "current": null,
  },
  "context": undefined,
  "eventEmitter": EventEmitter {
    "events": Object {},
  },
  "props": Object {
    "readOnly": false,
    "spritemap": "icons.svg",
  },
  "refs": Object {},
  "state": Object {},
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
